tests/nn_test.py

```diff
@@ -37,7 +38,154 @@
 config.parse_flags_with_absl()
 
 
+@jtu.with_config(jax_legacy_prng_key="allow")
 class NNFunctionsTest(jtu.JaxTestCase):
+  @parameterized.parameters([
+      [jnp.bfloat16, False],
+      [jnp.bfloat16, True],
+      [jnp.float16, False],
+      [jnp.float16, True],
+  ])
+  @jtu.skip_on_flag("jax_skip_slow_tests", True)
+  def testDotProductAttentionInfer(self, dtype, use_bias):
+    if not jtu.test_device_matches(["gpu"]):
```

Comment:
There are two major problems with your test: you have no non-GPU tests, and you don't test `implementation=None`. I would use a product test over the dtypes along with `implementation='xla'` and `implementation='cudnn'`, and compare against `implementation=None`, which should work on every platform. And only skip cudnn on non-GPU.

Reply:
Done.